ui/components/NodeDetails.js

```diff
@@ -0,0 +1,206 @@
+import React from 'react';
+import { NoSsr, TableCell, } from '@material-ui/core';
+import { MuiThemeProvider } from '@material-ui/core/styles';
+import {  createTheme } from '@material-ui/core/styles';
+import MUIDataTable from "mui-datatables";
+import TableSortLabel from '@material-ui/core/TableSortLabel'
+
+class NodeDetails extends React.Component {
```

Comment:
Any reason why we're using a class based component here instead of a functional component?

Reply:
If it's not too much of a hassle please let's switch to a functional component😅.